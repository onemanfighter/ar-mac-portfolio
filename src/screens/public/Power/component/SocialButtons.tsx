import {
  GithubIcon,
  InstagramIcon,
  LinkedInIcon,
  MediumIcon,
  YoutubeIcon,
} from '@assets';
import { Box, IconButton, Text } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

const SocialButtons = () => {
  const { t } = useTranslation();
  return (
    <>
      <Text
        color={'white'}
        textAlign={'center'}
        fontSize={'sm'}
        pos={'absolute'}
        align={'center'}
        bottom={32}
        left={'22%'}
      >
        {t('PowerScreen.sidebar.socialLinks.title')}
      </Text>
      <Box position={'absolute'} bottom={20} paddingX={3}>
        <IconButton
          variant="outline"
          color="white"
          _hover={{
            bg: 'white',
            color: 'blue',
          }}
          marginLeft={4}
          mx={2}
          as={Link}
          to={'https://in.linkedin.com/in/amitrai1603'}
          target="_blank"
          transition={'all 0.5s'}
          icon={<LinkedInIcon />}
          aria-label="linkedin"
        />
        <IconButton
          variant="outline"
          color="white"
          as={Link}
          to={'https://github.com/onemanfighter'}
          target="_blank"
          _hover={{
            bg: 'white',
            color: 'black',
          }}
          mx={2}
          transition={'all 0.5s'}
          icon={<GithubIcon />}
          aria-label="github"
        />
        <IconButton
          variant="outline"
          color="white"
          as={Link}
          to={'https://amitraikwar.medium.com/'}
          target="_blank"
          _hover={{
            bg: 'white',
            color: 'black',
          }}
          mx={2}
          transition={'all 0.5s'}
          icon={<MediumIcon />}
          aria-label="medium"
        />
        <IconButton
          variant="outline"
          color="white"
          as={Link}
          to={'https://www.youtube.com/'}
          target="_blank"
          _hover={{
            bg: 'white',
            color: 'red',
          }}
          mx={2}
          transition={'all 0.5s'}
          icon={<YoutubeIcon />}
          aria-label="youtube"
        />
        <IconButton
          variant="outline"
          color="white"
          _hover={{
            bg: 'white',
            color: 'pink.500',
          }}
          mx={3}
          transition={'all 0.5s'}
          icon={<InstagramIcon />}
          aria-label="instagram"
        />
      </Box>
      <Text
        color={'white'}
        textAlign={'center'}
        fontSize={'sm'}
        pos={'absolute'}
        align={'center'}
        bottom={8}
        left={'20%'}
      >
        {t('PowerScreen.sidebar.bottomText')}
      </Text>
      <Text
        color={'white'}
        textAlign={'center'}
        fontSize={'sm'}
        pos={'absolute'}
        align={'center'}
        bottom={1}
        left={'8%'}
      >
        {t('PowerScreen.sidebar.copyRight')}
      </Text>
    </>
  );
};

export default SocialButtons;
