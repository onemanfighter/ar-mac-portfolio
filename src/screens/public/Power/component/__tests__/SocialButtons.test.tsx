import { render } from '@testing-library/react';
import SocialButtons from '../SocialButtons';
import { BrowserRouter } from 'react-router-dom';

describe('SocialButtons', () => {
  it('should render correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <SocialButtons />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
