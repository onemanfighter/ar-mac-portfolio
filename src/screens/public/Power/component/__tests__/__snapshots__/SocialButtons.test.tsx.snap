// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButtons should render correctly 1`] = `
<div>
  <p
    class="chakra-text css-kzf5k3"
  >
    --------Social links---------
  </p>
  <div
    class="css-139ogue"
  >
    <a
      aria-label="linkedin"
      class="chakra-button css-1oe6ui6"
      href="https://in.linkedin.com/in/amitrai1603"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M0 1.146C0 .513.526 0 1.175 0h13.65C15.474 0 16 .513 16 1.146v13.708c0 .633-.526 1.146-1.175 1.146H1.175C.526 16 0 15.487 0 14.854V1.146zm4.943 12.248V6.169H2.542v7.225h2.401zm-1.2-8.212c.837 0 1.358-.554 1.358-1.248-.015-.709-.52-1.248-1.342-1.248-.822 0-1.359.54-1.359 1.248 0 .694.521 1.248 1.327 1.248h.016zm4.908 8.212V9.359c0-.216.016-.432.08-.586.173-.431.568-.878 1.232-.878.869 0 1.216.662 1.216 1.634v3.865h2.401V9.25c0-2.22-1.184-3.252-2.764-3.252-1.274 0-1.845.7-2.165 1.193v.025h-.016a5.54 5.54 0 01.016-.025V6.169h-2.4c.03.678 0 7.225 0 7.225h2.4z"
        />
      </svg>
    </a>
    <a
      aria-label="github"
      class="chakra-button css-ey8rer"
      href="https://github.com/onemanfighter"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0016 8c0-4.42-3.58-8-8-8z"
        />
      </svg>
    </a>
    <a
      aria-label="medium"
      class="chakra-button css-ey8rer"
      href="https://amitraikwar.medium.com/"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 640 512"
        width="1em"
      >
        <path
          d="M180.5 74.262C80.813 74.262 0 155.633 0 256s80.819 181.738 180.5 181.738S361 356.373 361 256 280.191 74.262 180.5 74.262zm288.25 10.646c-49.845 0-90.245 76.619-90.245 171.095s40.406 171.1 90.251 171.1 90.251-76.619 90.251-171.1H559c0-94.503-40.4-171.095-90.248-171.095zm139.506 17.821c-17.526 0-31.735 68.628-31.735 153.274s14.2 153.274 31.735 153.274S640 340.631 640 256c0-84.649-14.215-153.271-31.742-153.271z"
        />
      </svg>
    </a>
    <a
      aria-label="youtube"
      class="chakra-button css-gtm1u"
      href="https://www.youtube.com/"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8.051 1.999h.089c.822.003 4.987.033 6.11.335a2.01 2.01 0 011.415 1.42c.101.38.172.883.22 1.402l.01.104.022.26.008.104c.065.914.073 1.77.074 1.957v.075c-.001.194-.01 1.108-.082 2.06l-.008.105-.009.104c-.05.572-.124 1.14-.235 1.558a2.007 2.007 0 01-1.415 1.42c-1.16.312-5.569.334-6.18.335h-.142c-.309 0-1.587-.006-2.927-.052l-.17-.006-.087-.004-.171-.007-.171-.007c-1.11-.049-2.167-.128-2.654-.26a2.007 2.007 0 01-1.415-1.419c-.111-.417-.185-.986-.235-1.558L.09 9.82l-.008-.104A31.4 31.4 0 010 7.68v-.123c.002-.215.01-.958.064-1.778l.007-.103.003-.052.008-.104.022-.26.01-.104c.048-.519.119-1.023.22-1.402a2.007 2.007 0 011.415-1.42c.487-.13 1.544-.21 2.654-.26l.17-.007.172-.006.086-.003.171-.007A99.788 99.788 0 017.858 2h.193zM6.4 5.209v4.818l4.157-2.408L6.4 5.209z"
        />
      </svg>
    </a>
    <button
      aria-label="instagram"
      class="chakra-button css-1bjbnun"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="none"
        focusable="false"
        height="1em"
        viewBox="0 0 15 15"
        width="1em"
      >
        <path
          d="M7.5 5a2.5 2.5 0 100 5 2.5 2.5 0 000-5z"
          fill="currentColor"
        />
        <path
          clip-rule="evenodd"
          d="M4.5 0A4.5 4.5 0 000 4.5v6A4.5 4.5 0 004.5 15h6a4.5 4.5 0 004.5-4.5v-6A4.5 4.5 0 0010.5 0h-6zM4 7.5a3.5 3.5 0 117 0 3.5 3.5 0 01-7 0zM11 4h1V3h-1v1z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
  <p
    class="chakra-text css-1gjjbyc"
  >
    Made with ❤️ by Amit Raikwar
  </p>
  <p
    class="chakra-text css-1mqelpb"
  >
    © 2024 Amit Raikwar. All rights reserved.
  </p>
</div>
`;
